metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -556,7 +644,39 @@ func (sink *StackdriverSink) getResourceLabels(labels map[string]string) map[str
 	}
 }
 
-func createTimeSeries(resourceLabels map[string]string, metadata *metricMetadata, point *sd_api.Point) *sd_api.TimeSeries {
+func (sink *StackdriverSink) getContainerResourceLabels(labels map[string]string) map[string]string {
+	return map[string]string{
+		"project_id":     sink.project,
+		"location":       sink.zone,
+		"cluster_name":   sink.cluster,
+		"namespace_name": labels[core.LabelNamespaceName.Key],
+		"node_name":      labels[core.LabelNodename.Key],
+		"pod_name":       labels[core.LabelPodName.Key],
+		"container_name": labels[core.LabelContainerName.Key],
+	}
+}
+
+func (sink *StackdriverSink) getPodResourceLabels(labels map[string]string) map[string]string {
+	return map[string]string{
+		"project_id":     sink.project,
+		"location":       sink.zone,
+		"cluster_name":   sink.cluster,
+		"namespace_name": labels[core.LabelNamespaceName.Key],
+		"node_name":      labels[core.LabelNodename.Key],
+		"pod_name":       labels[core.LabelPodName.Key],
+	}
+}
+
+func (sink *StackdriverSink) getNodeResourceLabels(labels map[string]string) map[string]string {
+	return map[string]string{
+		"project_id":   sink.project,
+		"location":     sink.zone,
+		"cluster_name": sink.cluster,
+		"node_name":    labels[core.LabelNodename.Key],
+	}
+}
+
+func legacyCreateTimeSeries(resourceLabels map[string]string, metadata *metricMetadata, point *sd_api.Point) *sd_api.TimeSeries {
```

Comment:
why not implement this by calling createTimeSeries an passing "gke_container" as resource?